great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -101,6 +103,18 @@ def _save_project_config(self) -> None:
                         f"Saving {len(self.zep_config.datasources)} ZEP Datasources to {config_filepath}"
                     )
                     zep_json_dict: dict[str, JSONValues] = self.zep_config._json_dict()
+                    # re-combine the original raw config to preserve the original config ${FOO} placeholders
+                    # NOTE: this could cause issues depending on what values have the placeholders
+                    # TODO: Only update fields that had config placeholders?
+                    # TODO: cleanup
+                    updated_xdatasources = deep_update(
```

Comment:
If a user deletes parts of the config could this deep_update restore them accidentally? I can't seem to find the documentation for it.

Reply:
That's a great question. I assume I'll have to account for that.